feat: [CDS-70736]: Add OCI support, ECR AWS configuration (#2179)
harness/developer-hub: docs/continuous-delivery/gitops/add-a-harness-git-ops-repository.md

## Supported platforms

> * Currently, only Git platforms and HTTP Helm servers are supported in a GitOps Repository. Also OCI Helm repository is supported. 

Reviewer: ```suggestion
* Currently, only Git platforms, HTTP Helm servers, and OCI Helm repositories are supported in a GitOps Repository. 
```

        Harness will auto-detect the repository credentials template (if any) based on the GitOps repository **URL** and auto-populate it. If Harness auto-populated the GitOps repository, then you cannot edit the repository credentials template setting.

```mdx-code-block
</TabItem>
<TabItem value="OCI Helm repository" label="OCI Helm repository">
```

1. In **Specify Repository Type**, select **Helm**.
2. In **Repository Name**, enter a name.
3. In **GitOps Agent**, select or create the Agent you want to use to fetch charts from this repo. For details, go to [Install a Harness GitOps Agent](install-a-harness-git-ops-agent.md).
4. In **Repository URL**, enter the URL to your OCI Helm repository. For example, `registry-1.docker.io/docker`.
5. Select **Enable OCI**
6. Select **Continue**.
7. In **Credentials**, select **Specify Credentials for Repository**.
   - In **Credentials**, in **Connection Type**, select **HTTPS**, the **Anonymous** option to add a public repository.

>    - To add a private repository, select the authentication option with username and password. 

Reviewer: ```suggestion
   - To add a private repository, select the authentication option with username and password, and enter the access token.
```

     This authentication is supported for Docker and GitHub. AWS and Google have short-lived tokens, and may not work as expected. For details on how to configure access to AWS ECR, go to [Store and access OCI Helm repository in private AWS ECR](./helm-oci-repository-aws-ecr.md).

> 8. Select **Save & Continue**. Harness validates the connection.

Reviewer 2: This step (save and continue) is missing from the other two tabs. Should it be the last step in those tabs too?

        Harness will auto-detect the Repository Credentials Template (if any) based on the GitOps Repository **URL** and auto-populate it. If Harness auto-populated the GitOps Repository, then you cannot edit the Repository Credentials Template setting.

## Option: OCI Helm Repository
1. Click **Helm**.
2. In **Repository Name**, enter a name.
3. In **GitOps Agent**, select or create the Agent you want to use to fetch charts from this repo. See [Install a Harness GitOps Agent](install-a-harness-git-ops-agent.md).
4. In **Repository URL**, enter the URL to your OCI Helm Repository. For example, `registry-1.docker.io/docker`.

> 5. Clieck on **Enable OCI**

Reviewer 3: typo: Click
